src/ScopedValues.jl

```diff
@@ -63,17 +63,33 @@ Base.isassigned(val::ScopedValue) = val.has_default
 struct Snapshot
     values::Vector{Pair{ScopedValue, Any}}
 end
-Snapshot(values::ScopedValue...) = Snapshot(collect(map(v->v => v[], values)))
 
 """
-    @snapshot exprs...
+    snapshot()
+    snapshot(pairs::Pair{<:ScopedValue}...)
+    snapshot(values::ScopedValue...)
 
-Create a snapshot containing the `ScopedValue`(s) in `exprs`. This snapshot can
-be used in [`with`](@ref) or [`@with`](@ref) calls.
+Create a snapshot containing all the `ScopedValue`s in the current scope. If
+`pairs` or `values` are specified, they are added to the snapshot. This
+snapshot can be used in subsequent [`with`](@ref) or [`@with`](@ref) calls.
 """
-macro snapshot(exprs...)
-    exprs = map(esc, exprs)
-    :($(Snapshot)($(exprs...)))
+function snapshot()
+    values = Vector{Pair{ScopedValue, Any}}()
+    scope = current_scope()
+    if scope !== nothing
+        append!(values, collect(scope.values))
```

Comment:
This I am not so sure about since you might now snapshot ScopeValues you don't own.

Reply:
For our `TaskGroup` use case, we do want the ability to snapshot the current scope.

I could change the other `snapshot` methods to not include the current scope by default and add `append!`s for `Snapshot`s if you prefer?

Reply:
`scope.values` is accessing implementation specific internals.

Reply:
That's happening [here](https://github.com/vchuravy/ScopedValues.jl/blob/main/src/ScopedValues.jl#L76) also, to the same degree? It should be okay for the package (or for the `Base` implementation in 1.11+) to access implementation-specific internals I would think. We aren't exposing `Scope`s to the user, only `Snapshot`s.

Reply:
Yes this package owns Scope and accessing internals is fine, but it doesn't own Scope anymore on Julia 1.11.

I also don't think that a Snapshot by default should contain the entire scope. The idea behind ScopeValues is that you can only access the contents given the scoped value itself, that's the base contract.

Reply:
Once we have agreed on the new functionality here, I'll open a PR to add it upstream.

If we do not snapshot the current scope, then when there are multiple packages in use and each might specify some control via a scoped value, we would have to identify and enumerate all of these to snapshot them. This is inflexible and vulnerable to breakage.

Let's discuss on Slack.